Userland/Games/Flood/SettingsDialog.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "SettingsDialog.h"
+#include <AK/IntegralMath.h>
+#include <AK/QuickSort.h>
+#include <Games/Flood/SettingsDialogGML.h>
+#include <LibCore/DirIterator.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Button.h>
+#include <LibGUI/CheckBox.h>
+#include <LibGUI/ComboBox.h>
+#include <LibGUI/ItemListModel.h>
+#include <LibGUI/Label.h>
+#include <LibGUI/SpinBox.h>
+
+SettingsDialog::SettingsDialog(GUI::Window* parent, size_t board_rows, size_t board_columns, StringView color_scheme)
+    : GUI::Dialog(parent)
+    , m_board_rows(board_rows)
+    , m_board_columns(board_columns)
+    , m_color_scheme(color_scheme)
+{
+    set_rect({ 0, 0, 250, 150 });
+    set_title("New Game");
+    set_icon(parent->icon());
+    set_resizable(false);
```

Comment:
Still, this can mostly be in the GML, we have extensive documentation. Please don't leave a property setter in a widget constructor unless it doesn't have a GML accessor.

Reply:
okay I just misread those as widget setters, nevermind